File: Java/creational_patterns/src/Prototype/PhonebookFactory.java
package Prototype;

public class PhonebookFactory {
    private Phonebook phonebook;

    public PhonebookFactory(Phonebook phonebook) {
        this.phonebook = phonebook;
    }

    public void setPhonebook(Phonebook phonebook) {
        this.phonebook = phonebook;
    }

    public Phonebook copyPhonebook() {
        return (Phonebook) this.phonebook.copy();
    }
}

File: C#/CreationalPatterns/CreationalPatterns.Prototype/PhonebookFactory.cs
namespace CreationalPatterns.Prototype
{
    internal class PhonebookFactory
    {
        private Phonebook phonebook;

        public PhonebookFactory(Phonebook phonebook)
        {
            this.phonebook = phonebook;
        }

        public void SetPhonebook(Phonebook phonebook)
        {
            this.phonebook = phonebook;
        }

        public Phonebook CopyPhonebooke()
        {
            return (Phonebook)this.phonebook.Copy();
        }
    }
}
